### PR TITLE
Fixes #164 | Check if port index fits in bounds

### LINKED ISSUE #164: Port definition out of range
We've hit issues that we thought was resolved by https://github.com/allegro/marathon-consul/pull/145 one more time in different part of code [github.com/allegro/marathon-consul/apps/app.go:123](https://github.com/allegro/marathon-consul/blob/1.1.0/apps/app.go#L123)

```go
goroutine 8 [running]:
panic(0x7b8fe0, 0xc420010090)
        runtime/panic.go:500 +0x1a1
github.com/allegro/marathon-consul/apps.(*App).RegistrationIntents(0xc4205c3260, 0xc421738e00, 0x825b7c, 0x1, 0x4, 0xc420864af0, 0xc)
        github.com/allegro/marathon-consul/apps/app.go:123 +0x39d
github.com/allegro/marathon-consul/consul.(*Consul).marathonTaskToConsulServices(0xc4201642a0, 0xc421738e00, 0xc4205c3260, 0x0, 0x0, 0xc4202f3b20, 0x4ca5ba, 0xc42007a240)
        github.com/allegro/marathon-consul/consul/consul.go:284 +0x149
github.com/allegro/marathon-consul/consul.(*Consul).Register(0xc42
01642a0, 0xc421738e00, 0xc4205c3260, 0x1, 0xc420864750)
        github.com/allegro/marathon-consul/consul/consul.go:150 +0x7c
github.com/allegro/marathon-consul/sync.(*Sync).registerAppTasksNotFoundInConsul(0xc42000f360, 0xc420a2e000, 0x704, 0x95e, 0xc4217a8000, 0x8c7, 0xc00)
        github.com/allegro/marathon-consul/sync/sync.go:157 +0x255
github.com/allegro/marathon-consul/sync.(*Sync).syncServices(0xc42000f360, 0x5415e2, 0x5891a386)
        github.com/allegro/marathon-consul/sync/sync.go:78 +0x31b
github.com/allegro/marathon-consul/sync.(*Sync).SyncServices.func1()
        github.com/allegro/marathon-consul/sync/sync.go:56 +0x33
github.com/allegro/marathon-consul/vendor/github.com/rcrowley/go-metrics.(*StandardTimer).Time(0xc420230150, 0xc4200e73e0)
        github.com/allegro/marathon-consul/vendor/github.com/rcrowley/go-metrics/timer.go:212 +0x47
github.com/allegro/marathon-consul/metrics.Time(0x82cae3, 0xd, 0xc4200e73e0)
        github.com/allegro/marathon-consul/metrics/metrics.go:29 +0x6e
github.com/allegro/marathon-consul/sync.(*Sync).SyncServices(0xc42000f360, 0x0, 0x0)
        github.com/allegro/marathon-consul/sync/sync.go:56 +0xbe
github.com/allegro/marathon-consul/sync.(*Sync).StartSyncServicesJob.func1(0xc42000f360, 0xc42005e780)
        github.com/allegro/marathon-consul/sync/sync.go:42 +0x4d
created by github.com/allegro/marathon-consul/sync.(*Sync).StartSyncServicesJob
        github.com/allegro/marathon-consul/sync/sync.go:50 +0x2e7
```

### PR DESCRIPTION
**Pull request by janisz**

### CODE REVIEW

**Comment by janisz** [comment]
This happens when you deployed service and then update it with different number of ports but you stopped deployment or you deployment has not finished yet.

Below are the steps to reproduce it:

<details>

* `PUT /v2/apps`
```json
[{
  "id": "bug/164",
  "cmd": "python -m SimpleHTTPServer $PORT0",
  "cpus": 0.1,
  "mem": 32,
  "disk": 0,
  "instances": 1,
  "healthChecks": [
    {
      "gracePeriodSeconds": 3,
      "intervalSeconds": 60,
      "timeoutSeconds": 20,
      "maxConsecutiveFailures": 3,
      "portIndex": 0,
      "path": "/",
      "protocol": "HTTP",
      "ignoreHttp1xx": false
    }
  ],
  "labels": {
    "consul": "name"
  },
  "portDefinitions": [
    {
      "protocol": "tcp"
    }
  ]
}]
```

* `PUT /v2/apps`
```json
[{
  "id": "bug/164",
  "cmd": "fail",
  "cpus": 0.1,
  "mem": 32,
  "disk": 0,
  "instances": 1,
  "healthChecks": [
    {
      "gracePeriodSeconds": 3,
      "intervalSeconds": 60,
      "timeoutSeconds": 20,
      "maxConsecutiveFailures": 3,
      "portIndex": 0,
      "path": "/",
      "protocol": "HTTP",
      "ignoreHttp1xx": false
    }
  ],
  "labels": {
    "consul": "name"
  },
  "portDefinitions": [
    {
      "protocol": "tcp"
    },
    {
      "protocol": "tcp"
    }
  ]
}]
```
* `GET /v2/apps/bug/164?embed=app.tasks`
```json
{
  "app":{
    "id":"/bug/164",
    "ports":[
      10005,
      10006
    ],
    "portDefinitions":[
      {
        "port":10005,
        "protocol":"tcp"
      },
      {
        "port":10006,
        "protocol":"tcp"
      }
    ],
    "tasks":[
      {
        "state":"TASK_RUNNING",
        "ports":[
          31018
        ],
        "id":"bug_164.adaf2c2c-e88d-11e6-ae5c-a44e31765720",
        "appId":"/bug/164"
      }
    ]
  }
}
```

</details>

**Comment by ojagodzinski** [comment]
Can service be processed anyway, despite the inconsistency between app definition and running task Totally 'ignoring' such cases may not be the best idea.

**Comment by janisz** [comment]
I concur. If port counts differs between task and app we can check version and if it differs to we can skip registration of whole task. This approach could work but does not prevent situation when port definitions totally changed but port counts stays the same.

File: apps/app_test.go [hunk 1]
```diff
@@ -331,6 +356,36 @@ func TestRegistrationIntent_MultipleIntentsViaPortDefinitionIfMultipleContainCon
 	assert.Equal(t, []string{"common-tag", "second-tag"}, intents[1].Tags)
 }
 
+func TestRegistrationIntent_TaskHasLessPortsThenApp(t *testing.T) {
```
Comment (dankraw) [inline apps/app_test.go]: Then -> Than ;)

Comment (janisz) [inline apps/app_test.go]: Good catch

**Pull request merged**